### PR TITLE
ModuleBase fix

### PR DESCRIPTION
**Pull request by chlupac**
Fix of #252 and some cosmetics

### CODE REVIEW

File: Oqtane.Client/Modules/ModuleBase.cs [hunk 1]
```diff
@@ -10,11 +10,12 @@ namespace Oqtane.Modules
 {
     public class ModuleBase : ComponentBase, IModuleControl
     {
-        public Logger logger { get; set; }
+        private Logger _logger;
+
+        protected Logger logger => _logger ?? (_logger = new Logger(this));
```
Comment (chlupac) [inline Oqtane.Client/Modules/ModuleBase.cs]: Lazy loading of logger, it was instantiated on every module loaded from database

File: Oqtane.Client/Modules/ModuleBase.cs [hunk 2]
```diff
@@ -10,11 +10,12 @@ namespace Oqtane.Modules
...
 
         public ModuleBase()
         {
-            logger = new Logger(this);
```
Comment (chlupac) [inline Oqtane.Client/Modules/ModuleBase.cs]: Ah, I forgot delete this

**Comment by sbwalker** [comment]
Can you explain why it is necessary to do lazy loading? Since this code was working fine before, I would assume that a functional change would not be required to resolve an issue caused by renaming variables?

**Comment by chlupac** [comment]
It is not necessary, it's optimization and cosmetic. You instantiated all modules during boot, not only used ones. Constructor was called for all modules. I can delete it if you want, it's my coding style how I solve this patterns.

**Comment by sbwalker** [comment]
Sorry... it is difficult to do code reviews when fixes and enhancements are included in the same pull request. I was expecting that the PR was to fix a bug introduced by the renaming, so the other code change did not make sense to me. However it is a good optimization so I will merge it.

**Pull request merged**